src/main/java/net/neoforged/neoforge/fluids/FluidType.java

```diff
@@ -535,6 +552,27 @@ public int getViscosity(FluidState state, BlockAndTintGetter getter, BlockPos po
         return this.getViscosity();
     }
 
+    /**
+     * Returns whether a fluid above a pointed dripstone block can successfully fill a cauldron below.
+     *
+     * @param fluid the fluid that is dripping from a dripstone
+     * @param level the level the fluid is being placed in
+     * @param pos   the position of the cauldron this fluid is dripping into
+     * @return {@code true} if a cauldron is successfully filled, {@code false} otherwise
+     */
+    public boolean handleDripInfo(Fluid fluid, Level level, BlockPos pos) {
+        if (fluid.isSource(fluid.defaultFluidState()) && fluid.getFluidType().getDripInfo() != null) {
```

Comment:
If we can't do this check normally, we should just check whether the fluid is an instance of `FlowingFluid` and compare the source block via `#getSource`. If somebody wants to use a non-flowing fluid for this, they can just reimplement the method.

Reply:
I think I did what you wanted here? Let me know if you had something else in mind, I'm happy to adjust as needed